DXMainClient/DXGUI/Multiplayer/GameLobby/CnCNetGameLobby.cs

```diff
@@ -755,9 +778,16 @@ private void HandleOptionsRequest(string playerName, int options)
             if (color < 0 || color > MPColors.Count)
                 return;
 
-            var disallowedSides = GetDisallowedSides();
+            // Disallowed sides from client, maps, or game modes do not take random selectors into account
+            // So, we need to insert "false" for each random at the beginning of this list AFTER getting them
+            // from client, maps, or game modes.
+            var randomDisallowedSides = new List<bool>(RandomSelectorCount);
+            for (int i = 0; i < RandomSelectorCount; i++)
+                randomDisallowedSides.Add(false);
+
+            var disallowedSides = randomDisallowedSides.Concat(GetDisallowedSides()).ToArray();
 
-            if (side > 0 && side <= SideCount && disallowedSides[side - 1])
+            if (0 < side && side < SideCount && disallowedSides[side])
```

Comment:
I do not quite understand what is happening here, perhaps there should be some rephrasing done?